### PR TITLE
Add JSONPath + example for ConditionalInterceptor 

### PR DESCRIPTION
**Pull request by christiangoerdes**
- Implement JSONPath functionality in BuiltInFunctions to evaluate expressions using JSON data.
- Add test cases for the new JSONPath function within BuiltInFunctionsTest.
- Introduce conditional logic in the API configuration to handle JSON requests based on specific criteria, using Spring Expression Language (SpEL).

### CODE REVIEW

**Comment by rrayst** [comment]
This branch is currently /not-ok-to-test .

Please merge the master branch into this branch (e.g. 'Update branch' above) and resolve the conflicts. Once completed, feel free to set /ok-to-test again.

File: core/src/test/java/com/predic8/membrane/core/lang/spel/functions/BuiltInFunctionsTest.java [hunk 1]
```diff
@@ -36,9 +36,16 @@ public class BuiltInFunctionsTest {
     static void init() throws URISyntaxException {
         var exc = Request.get("foo").buildExchange();
         exc.setProperty(SECURITY_SCHEMES, List.of(new ApiKeySecurityScheme(HEADER, "X-Api-Key").scopes("demo", "test")));
+        exc.getRequest().setBodyContent("{\"name\":\"John\"}".getBytes());
```
Comment (predic8) [inline core/src/test/java/com/predic8/membrane/core/lang/spel/functions/BuiltInFunctionsTest.java]: Mutliline String

File: core/src/main/java/com/predic8/membrane/core/interceptor/flow/ConditionalInterceptor.java [hunk 2]
```diff
@@ -63,7 +63,7 @@ public class ConditionalInterceptor extends AbstractFlowInterceptor {
     /**
      * Spring Expression Language
      */
-    private final SpelParserConfiguration spelConfig = new SpelParserConfiguration(SpelCompilerMode.IMMEDIATE, this.getClass().getClassLoader());
+    private final SpelParserConfiguration spelConfig = new SpelParserConfiguration(SpelCompilerMode.MIXED, this.getClass().getClassLoader());
```
Comment (predic8) [inline core/src/main/java/com/predic8/membrane/core/interceptor/flow/ConditionalInterceptor.java:68]: Comment why MIXED

File: core/src/main/java/com/predic8/membrane/core/interceptor/flow/ConditionalInterceptor.java [hunk 3]
```diff
@@ -62,6 +62,7 @@ public class ConditionalInterceptor extends AbstractFlowInterceptor {
 
     /**
      * Spring Expression Language
+     * SpEL configuration with MIXED mode allows both interpreted and compiled expression evaluation.
```
Comment (predic8) [inline core/src/main/java/com/predic8/membrane/core/interceptor/flow/ConditionalInterceptor.java:65]: Missing: The reason why we used this - Like didn't work with X

**Review by predic8 — CHANGES_REQUESTED** [review]

**Review by predic8 — APPROVED** [review]

**Pull request merged**